CHANGES.md

```diff
@@ -36,6 +36,7 @@
 
 * ReadFromPubSub(topic=<topic>) in Python previously created a subscription under the same project as the topic. Now it will create the subscription under the project specified in pipeline_options. If the project is not specified in pipeline_options, then it will create the subscription under the same project as the topic. ([BEAM-3453](https://issues.apache.org/jira/browse/BEAM-3453)).
 * SpannerAccessor in Java is now package-private to reduce API surface. `SpannerConfig.connectToSpanner` has been moved to `SpannerAccessor.create`. ([BEAM-9310](https://issues.apache.org/jira/browse/BEAM-9310)).
+* PCollections will now have their tags correctly propagated through the Pipeline. Users may expect the old implementation which gave PCollection output ids a monotonically increasing id. To go back to the old implementation, use the "force_generated_pcollection_output_ids" experiment. The default is the new implementation (force_generated_pcollection_output_ids=False).
```

Comment:
Are you sure, that we would like to change the default in the first release of the change?

Reply:
I looked into your question a bit more about determinism in the generated ids, and I found that since it traverses dicts, the order is non-deterministic. DoOutputsTuple manually add themselves correctly to their producer. For Tuples, I'm keeping the old implementation. For PValues, I fixed the bug to now correctly propagate the tag. 

Unless I'm mistaken, for the user to want to use the old implementation they are then either:

- Relying on a bug (PValue)
- Relying on non-deterministic behavior for generated tags (dicts)
- Or using Tuples, which I didn't change.

Is this okay to leave the default as-is?

Reply:
What is the best practice for rolling out experiments in Beam? In any case, we should set it as the default at least within interactive runner.

Reply:
Sam, I would say that the existing behavior was a bug for the points that you mentioned or that the user never really cared what the tag names are (e.g. for a partition transform).

Reply:
With this explanation, I think it is fine to change the default in this release.

FWIW, dictionaries have deterministic iteration order for py >= 3.6.